module-apps/application-messages/models/SMSThreadModel.cpp

```diff
@@ -90,6 +92,15 @@ auto SMSThreadModel::handleQueryResponse(db::QueryResult *queryResult) -> bool
 
     number = std::make_unique<utils::PhoneNumber::View>(msgResponse->getNumber());
 
+    if (number) {
+        auto &threadNumber = static_cast<utils::PhoneNumber::View &>(*number);
+
+        DBServiceAPI::GetQuery(
+            application,
+            db::Interface::Name::Notifications,
+            std::make_unique<db::query::notifications::Increment>(NotificationsRecord::Key::Sms, threadNumber, -1));
```

Comment:
Sadly, this solution doesn't work as expected: notifications are added for each new (unread) message, but the clearing of notifications has to occur for each thread - and we need to know how many unread messages are there in a thread. We don't keep that information.

Reply:
AFAIR this information is stored in different data base. Please take a look at ThreadRecord and unreadMsgCount.

Reply:
We figured this eventually after discussing with @PrzeBrudny

Reply:
Resolved